### PR TITLE
Merging linear streams

### LINKED ISSUE #162: Merge linear-streams and migrate issues
Merge (via git-subtree-ing) an optimized, documented version of linear-streams (with sufficient workarounds for the API that can't be implemented properly without affine streams, e.g., `readStdinUntil :: (String -> Bool) -> Stream (Of String) IO ()`).

### PR DESCRIPTION
**Pull request by Divesh-Otwani**
This PR moves linear-streams into linear-base. A migration of the issues with a tag will follow the merge.
All the changes in linear-streams have already been reviewed so this PR is about just checking if the merge is clean.
The linear-streams repository will be deleted following the merge.

I'll have a separate PR about moving the stuff from the readme of linear-streams somewhere in linear-base.

Depends on #200 and depends on #201. 
Finally closes #162.

### CODE REVIEW

File: linear-base.cabal [hunk 1]
```diff
@@ -43,6 +43,14 @@ library
...
     Foreign.Marshal.Pure
     Prelude.Linear
+    Streaming.Streaming
+    Streaming.Prelude
+    Streaming.Internal.Consume
+    Streaming.Internal.Interop
+    Streaming.Internal.Many
+    Streaming.Internal.Process
+    Streaming.Internal.Produce
+    Streaming.Internal.Type
```
[the remark this answers is not in the document]
Comment (Divesh-Otwani) [reply linear-base.cabal:53]: `Streaming` doesn't have more than two modules. I think `Streaming.Linear` and `Streaming.Prelude.Linear` is a good idea.

File: linear-base.cabal [hunk 2]
```diff
@@ -43,6 +43,14 @@ library
     Debug.Trace.Linear
     Foreign.Marshal.Pure
     Prelude.Linear
+    Streaming.Streaming
```
[the remark this answers is not in the document]
Comment (Divesh-Otwani) [reply linear-base.cabal:46]: Yup, agreed.

**Comment by aspiwack** [comment]
You got your base-branch wrong as far as dependencies go. But anyway, I'd suggest first importing everything with `git subtree`. Then a commit could merge the `stack.yaml`, then we think whether it should be in linear base or not (depending on whether we want to use streams in linear base or not, I suppose).

The point is to keep the history.

**Comment by Divesh-Otwani** [comment]
Closing to be replaced by #210.

**Pull request closed without merging**